File: src/types/fmg/data/controller/event/ChampionModelEvent.java
package fmg.data.controller.event;

import java.util.EventObject;

import fmg.core.model.mosaics.EMosaic;
import fmg.data.controller.serializable.ChampionsModel;
import fmg.data.controller.types.ESkillLevel;

public class ChampionModelEvent extends EventObject {
	private static final long serialVersionUID = 1L;

	public static final int POS_ALL = -1;
	public static final int INSERT = 1, UPDATE = 2, DELETE = 3;

	private final EMosaic mosaic;
	private final ESkillLevel skill;
	private final int pos;
	private final int type;
	
	public ChampionModelEvent(ChampionsModel source, EMosaic mosaic, ESkillLevel skill, int pos, int type) {
		super(source);
		this.mosaic = mosaic;
		this.skill = skill;
		this.pos = pos;
		this.type = type;
	}

	public ESkillLevel getSkill() {
		return skill;
	}
	public EMosaic getMosaic() {
		return mosaic;
	}
	public int getPos() {
		return pos;
	}
	public int getType() {
		return type;
	}

	@Override
	public ChampionsModel getSource() {
		return (ChampionsModel) super.getSource();
	}
}

File: src/src_data/fmg/data/controller/event/ChampionModelEvent.cs
using fmg.data.controller.serializable;
using fmg.data.controller.types;
using fmg.core.model.mosaics;

namespace fmg.data.controller.Event {

public class ChampionModelEvent {

   public delegate void OnChanged(ChampionsModel source, ChampionModelEvent _event);

   public const int POS_ALL = -1;
	public const int INSERT = 1, UPDATE = 2, DELETE = 3;

	private EMosaic mosaic;
	private ESkillLevel skill;
	private int pos;
	private int type;
	
	public ChampionModelEvent(EMosaic mosaic, ESkillLevel skill, int pos, int type) {
		this.mosaic = mosaic;
		this.skill = skill;
		this.pos = pos;
		this.type = type;
	}

	public ESkillLevel getSkill() {
		return skill;
	}
	public EMosaic getMosaic() {
		return mosaic;
	}
	public int getPos() {
		return pos;
	}
	public int getType() {
		return type;
	}
}
}